tsup.config.ts

```diff
@@ -3,14 +3,30 @@ import { promises as fsPromises } from "fs";
 import path from "path";
 import postcss from "postcss";
 import postcssModules from "postcss-modules";
+import { glob } from "glob";
 
 export default defineConfig({
-  entry: ["src/index.tsx"],
+  entry: glob.sync("src/**/*.{ts,tsx}", {
+    cwd: process.cwd(),
+    ignore: [
+      "**/node_modules/**",
+      "**/dist/**",
+      "**/cjs/**",
+      "**/esm/**",
+      "**/*.stories.*",
+      "**/*.test.*",
+    ],
+  }),
   format: ["cjs", "esm"],
   external: ["react", "react-dom"],
   noExternal: ["@cambly/syntax-design-tokens"],
   sourcemap: true,
   dts: true,
+  splitting: true,
+  skipNodeModulesBundle: true,
+  esbuildOptions(options) {
+    options.chunkNames = "__chunks/[hash]";
```

Comment:
@benjroy any idea how we can make these chunk names more readable? 
![image](https://github.com/Cambly/syntax/assets/127199/4d8d8e69-65f9-47dc-9a9c-adac017c77b6)

Reply:
if we leave this option off, our chunk files will be named things like this:

<img width="462" alt="image" src="https://github.com/Cambly/syntax/assets/1890801/0b41a39a-cfda-4429-aac7-65253597efa5">


i.e. that `[hash]` is always going to be in there somehow.  `splitting: true` is the default behavior in esbuild here, but if we set `splitting: false` then we won't have any of those chunks generated.  Instead we'll have a mix of `js` & `mjs` files matching the names of the source files:  

<img width="275" alt="image" src="https://github.com/Cambly/syntax/assets/1890801/3480de4c-ce6e-4338-8338-262dd2dcedcd">


The output is different though, so I'd like to explore that possibility in a separate PR as well, just like watching the impact of setting `sideEffects: true`.  How do you feel about that?